packages/editor/src/components/block-list/block.js

```diff
@@ -483,14 +483,17 @@ export class BlockListBlock extends Component {
 			>
 				{ ! isPartOfMultiSelection && isMovable && (
 					<BlockDraggable
-						rootClientId={ rootClientId }
-						index={ order }
-						clientId={ clientId }
-						layout={ layout }
+						elementId={ blockElementId }
+						transferData={ {
+							type: 'block',
+							fromIndex: order,
+							rootClientId,
+							clientId,
+							layout,
+						} }
```

Comment:
This is something that has leaked my next step :)

This PR is the first step in refactoring DnD. The idea is to get rid of the `BlockDraggable` component (an invisible drag handler) in favor of a visible drag affordance. Originally, the proposal was to use the hover label https://github.com/WordPress/gutenberg/issues/7114 but that had some problems and now the plan is to include it within the mover component. That would be my next PR after this is merged.

Unless there is a strong opposition I'd leave it as it is, and we can discuss improvements to that in the next PR.

Reply:
Actually, I think we should leave the `BlockDraggable` even if it becomes UI-less similarily to the `Draggable` component. It would inject the required `transfertData` prop to the `Draggable` component. That way we can easily have multiple `BlockDraggable` in different places. (Just pass the `uid`)

Reply:
I like that abstraction. Besides what you mention it also builds up nicely: we could have an `ImageDraggable` component whose only responsibility would be to take care of dragging images around, and so on.

At the moment, I've reverted that change in 8b77faa We can dig deeper into simplifying the input data for `BlockDraggable` in the next step when it becomes UI-less.